packages/driver/src/cypress/command_queue.ts

```diff
@@ -394,15 +445,31 @@ export class CommandQueue extends Queue<$Command> {
         err.name = 'CypressError'
       }
 
-      commandRunningFailed(Cypress, this.state, err)
+      const current = this.state('current')
+
+      commandRunningFailed(Cypress, err, current)
+
+      if (err.isRecovered) {
+        current?.recovered()
+
+        return // let the queue end & restart on to the next command index (set in onQueueFailed)
+      }
+
+      if (current?.state === 'queued') {
+        current.skip()
+      } else if (current?.state === 'pending') {
+        current.fail()
+      }
+
+      this.cleanup()
 
       return this.fail(err)
     }
 
     const { promise, reject, cancel } = super.run({
       onRun: next,
       onError,
-      onFinish: this.cleanup,
+      onFinish: this.run,
```

Comment:
The 2 layers of recursion are messing with me a bit. Could we pass `next` here instead and achieve the same thing?

Reply:
Its def weird, however because we are storing the promise, reject & cancel func in state to bail out, if we pass `next` here instead, we don't create another promise chain to handle further errors:

When passing `next` | when passing `this.run`
--|--
<img width="679" alt="Screen Shot 2022-11-04 at 2 08 26 PM" src="https://user-images.githubusercontent.com/14099737/200056474-63b16ea4-6a41-421d-877b-7c37d539069a.png">|<img width="646" alt="Screen Shot 2022-11-04 at 2 08 06 PM" src="https://user-images.githubusercontent.com/14099737/200056478-0b5a6da2-9b3f-465c-929e-b01dbcd04e15.png">